test/src/test/java/org/corfudb/runtime/view/ManagementViewTest.java

```diff
@@ -1516,4 +1517,104 @@ public void queryClusterStatus() throws Exception {
+     * Scenario: Cluster of 2 nodes - Nodes 0 and 1
+     * Some data (10 appends) is written into the cluster.
+     * Then rules are added on both the nodes' management agents so that they cannot reconfigure
+     * the system. Another rule is added to the tail of the chain to drop all READ_RESPONSES.
+     * The epoch is incremented and the new layout is pushed to both the nodes.
+     * NOTE: The sequencer is not bootstrapped for the new epoch.
+     * Now, both the management agents attempt to bootstrap the new sequencer but the
+     * FastObjectLoaders should stall due to the READ_RESPONSE drop rule.
+     * This triggers the systemDownHandler.
+     */
+    @Test
+    public void triggerSystemDownHandlerInDeadlock() throws Exception {
+        addServer(SERVERS.PORT_0);
+        addServer(SERVERS.PORT_1);
+
+        Layout l = new TestLayoutBuilder()
+                .setEpoch(1L)
+                .addLayoutServer(SERVERS.PORT_0)
+                .addLayoutServer(SERVERS.PORT_1)
+                .addSequencer(SERVERS.PORT_0)
+                .addSequencer(SERVERS.PORT_1)
+                .buildSegment()
+                .buildStripe()
+                .addLogUnit(SERVERS.PORT_0)
+                .addLogUnit(SERVERS.PORT_1)
+                .addToSegment()
+                .addToLayout()
+                .setClusterId(UUID.randomUUID())
+                .build();
+        bootstrapAllServers(l);
+        corfuRuntime = getRuntime(l).connect();
+
+        CorfuRuntime managementRuntime0 = getManagementServer(SERVERS.PORT_0)
+                .getManagementAgent().getCorfuRuntime();
+        CorfuRuntime managementRuntime1 = getManagementServer(SERVERS.PORT_1)
+                .getManagementAgent().getCorfuRuntime();
+
+        // Waiting for management servers to send the bootstrap sequencer request and be ready
+        // to detect failures to speed up test time.
+        CFUtils.within(
+                CompletableFuture.allOf(
+                        getManagementServer(SERVERS.PORT_0).getManagementAgent().getRecoveryBarrierFuture(),
+                        getManagementServer(SERVERS.PORT_1).getManagementAgent().getRecoveryBarrierFuture()),
+                PARAMETERS.TIMEOUT_NORMAL
+        ).join();
+
+        IStreamView streamView = corfuRuntime.getStreamsView()
+                .get(CorfuRuntime.getStreamID("testStream"));
+        final byte[] payload = "test".getBytes();
+        final int num = 10;
+        for (int i = 0; i < num; i++) {
+            streamView.append(payload);
+        }
+
+        // Setting aggressive timeouts
+        setAggressiveTimeouts(l, corfuRuntime, managementRuntime0, managementRuntime1);
+
+        setAggressiveDetectorTimeouts(SERVERS.PORT_0, SERVERS.PORT_1);
+
+        final Semaphore semaphore = new Semaphore(2);
+        semaphore.acquire(2);
+
+        //Release the semaphore on invocation.
+        Runnable systemDownHandler = () -> {
```

Comment:
shouldn't this also test the actual handler? why redefine another one just for the test?

Reply:
You are right, I just wanted a way to know that this handler was invoked.
If I remove this, this test will become a black-box test and I will have to rely and assert on the fact that the detector is not stuck and handles some failures. Is that reasonable?